reference/docs-conceptual/learn/ps101/05-formatting-aliases-providers-comparison.md

```diff
@@ -0,0 +1,651 @@
+
+```powershell
+$Numbers -notcontains 15
+```
+
+```Output
+True
+```
+
+The previous example returns the **Boolean** true because it's true that the `$Numbers` variable
+doesn't contain 15. It does however contain the number 10 so it's false when it's tested.
+
+```powershell
+$Numbers -notcontains 10
+```
+
+```Output
+False
+```
+
+The "in" comparison operator was first introduced in PowerShell version 3.0. It's used to determine
+if a value is "in" an array. The `$Numbers` variable is an array since it contains multiple values.
+
+```powershell
+15 -in $Numbers
+```
+
+```Output
+False
+```
+
+In other words, `-in` performs the same test as the contains comparison operator except from the
+opposite direction.
+
+```powershell
+10 -in $Numbers
+```
+
+```Output
+True
+```
+
+15 isn't in the `$Numbers` array so false is returned in the following example.
+
+```powershell
+15 -in $Numbers
+```
+
+```Output
+False
+```
+
+Just like the `-contains` operator, `not` reverses the logic for the `-in` operator.
+
+```powershell
+10 -notin $Numbers
+```
+
+```Output
+False
+```
+
+The previous example returns false because the `$Numbers` array does include 10 and the condition
+was testing to determine if it didn't contain 10.
+
+15 is "not in" the `$Numbers` array so it returns the **Boolean** true.
+
+```powershell
+15 -notin $Numbers
+```
+
+```Output
+True
+```
+
+The `-replace` operator does just want you would think. It's used to replace something. Specifying
+one value replaces that value with nothing. In the following example, I replace "Shell" with nothing.
+
+```powershell
+'PowerShell' -replace 'Shell'
+```
+
+```Output
+Power
+```
+
+If you want to replace a value with a different value, specify the new value after the pattern you
+want to replace. SQL Saturday in Baton Rouge is an event that I try to speak at every year. In the
+following example, I replace the word "Saturday" with the abbreviation "Sat".
+
+```powershell
+'SQL Saturday - Baton Rouge' -Replace 'saturday','Sat'
+```
+
+```Output
+SQL Sat - Baton Rouge
+```
+
+There are also methods like **Replace()** that can be used to replace things similar to the way the
+replace operator works. However, the `-Replace` operator is case-insensitive by default, and the
+**Replace()** method is case-sensitive.
+
+```powershell
+'SQL Saturday - Baton Rouge'.Replace('saturday','Sat')
+```
+
+```Output
+SQL Saturday - Baton Rouge
+```
+
+Notice that the word "Saturday" wasn't replaced in the previous example. This is because it was
+specified in a different case than the original. When the word "Saturday" is specified in the same
+case as the original, the **Replace()** method does replace it as expected.
+
+```powershell
+'SQL Saturday - Baton Rouge'.Replace('Saturday','Sat')
+```
+
+```Output
+SQL Sat - Baton Rouge
+```
+
+Be careful when methods to transform data because you can run into unforeseen problems, such as
+failing the _Turkey Test_. For an example, see the blog article titled
```

Comment:
```suggestion
Be careful when using methods to transform data because you can run into unforeseen problems, such as
failing the _Turkey Test_. For an example, see the blog article titled
```

Reply:
Fixed